### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by bpoplauschi**
Initial version of the unit tests setup:
- Using [OCMock](https://github.com/erikdoe/ocmock) for mocking and stubbing, [Expecta](https://github.com/specta/expecta) to easily set expectations and the **CocoaLumberjack** framework via local pod
- added some basic tests to demonstrate how to write tests for our library and to make sure the basic functionality works.

### CODE REVIEW

**Comment by bpoplauschi** [comment]
@rivera-ernesto please take a look. I still need to add some real tests, but I want your opinion from a setup perspective.

**Comment by bpoplauschi** [comment]
@rivera-ernesto I managed to get to a working setup for the tests. I added a few basic sanity checks, from now on we can take it from there. Could you please merge this one in so I can start working on the 2.0.0 release?

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +18,13 @@ script:
...
     - xctool -project Framework/Mobile/Lumberjack.xcodeproj -scheme 'LibTest' -configuration Release -sdk iphonesimulator -arch i386 build
     - xctool -project Framework/Desktop/Lumberjack.xcodeproj -scheme 'CocoaLumberjack'
-#comment    - xctool -project Framework/Desktop/Lumberjack.xcodeproj -scheme 'CocoaLumberjack' -configuration Release -sdk iphonesimulator -arch i386 build
+    - xctool -project Framework/Desktop/Lumberjack.xcodeproj -scheme 'CocoaLumberjack' -configuration Release -sdk iphonesimulator -arch i386 build
+
+    - cd Tests
+    - pod install
+    - cd ..
+    - xctool -workspace Framework/Lumberjack.xcworkspace -scheme 'iOS Tests' -sdk iphonesimulator -arch i386 test
+    - xctool -workspace Framework/Lumberjack.xcworkspace -scheme 'OS X Tests' test
```
Comment (rivera-ernesto) [inline .travis.yml]: Looks good, but those are only building the tests and nor running them right?

Comment (bpoplauschi) [inline .travis.yml]: Those are running the tests. See Travis logs

**Comment by bpoplauschi** [comment]
@rivera-ernesto please merge this :)

**Comment by bpoplauschi** [comment]
```
      ✓ Touch iOS Tests.xctest (12 ms)
      0 errored, 0 warning (3111 ms)
  [Info] Collecting info for testables... (2859 ms)
  run-test iOS Tests.xctest (iphonesimulator8.1, logic-test)
      -[DDBasicLoggingTests testAll5DefaultLevelsAsync]
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Invalid connection: com.apple.coresymbolicationd
2015-03-13 08:46:28:521 xctest[2374:303] Error
2015-03-13 08:46:28:586 xctest[2374:303] Warn
2015-03-13 08:46:28:587 xctest[2374:303] Info
2015-03-13 08:46:28:587 xctest[2374:303] Debug
2015-03-13 08:46:28:587 xctest[2374:303] Verbose
━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    ✓ -[DDBasicLoggingTests testAll5DefaultLevelsAsync] (134 ms)
      -[DDBasicLoggingTests testLoggerLogLevelAsync]
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
2015-03-13 08:46:28:591 xctest[2374:303] Error
2015-03-13 08:46:28:593 xctest[2374:303] Warn
━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    ✓ -[DDBasicLoggingTests testLoggerLogLevelAsync] (7 ms)
      -[DDBasicLoggingTests testX_ddLogLevel_async]
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
2015-03-13 08:46:28:593 xctest[2374:303] Info
2015-03-13 08:46:28:593 xctest[2374:303] Debug
2015-03-13 08:46:28:593 xctest[2374:303] Verbose
2015-03-13 08:46:28:598 xctest[2374:303] Error
2015-03-13 08:46:28:607 xctest[2374:303] Warn
2015-03-13 08:46:28:607 xctest[2374:303] Info
━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    ✓ -[DDBasicLoggingTests testX_ddLogLevel_async] (10 ms)
    3 passed, 0 failed, 0 errored, 3 total (153 ms)
** TEST SUCCEEDED: 3 passed, 0 failed, 0 errored, 3 total ** (15838 ms)
```

**Comment by rivera-ernesto** [comment]
It looks great! Even GitHub shows 2 successful checks.

**Comment by rivera-ernesto** [comment]
Feel free to release 2.0.0 whenever you feel ready ;)

**Comment by bpoplauschi** [comment]
Great

**Comment by bpoplauschi** [comment]
It is done. Very nice :+1:

**Pull request merged**